src/main/java/org/hawkular/apm/qe/JaegerQEBase.java

```diff
@@ -0,0 +1,241 @@
+    public static Double JAEGER_SAMPLING_RATE = 1.0;
+    public static Integer JAEGER_PORT = 0;
+    public static String JAEGER_URL = System.getenv().getOrDefault("JAEGER_URL", "localhost");
+    public static String SERVICE_NAME = System.getenv()
+            .getOrDefault("SERVICE_NAME", "qe-automation");
+
+    public ObjectMapper jsonObjectMapper = new ObjectMapper();
+
+    /**
+     *
+     * @return
+     */
+    public Tracer getTracer() {
+        return getJaegerTracer(JAEGER_URL);
+    }
+
+
+    /**
+     *
+     * @param url
+     * @return
+     */
+    public Tracer getJaegerTracer(String url) {
+        _logger.info("creating tracer with url [{}]", url);
+        Sender sender = new UDPSender(url, JAEGER_PORT, JAEGER_MAX_PACKET_SIZE);
+        Metrics metrics = new Metrics(new StatsFactoryImpl(new NullStatsReporter()));
+        Reporter reporter = new RemoteReporter(sender, JAEGER_FLUSH_INTERVAL, JAEGER_MAX_QUEUE_SIZE, metrics );
+        Sampler sampler = new ProbabilisticSampler(JAEGER_SAMPLING_RATE);
+
+        Tracer tracer = new com.uber.jaeger.Tracer.Builder(SERVICE_NAME, reporter, sampler)
+                .build();
+
+        return tracer;
+    }
+
+
+    /**
+     * Make sure spans are flushed before trying to retrieve them
+     */
+    public void waitForFlush() {
+        try {
+            Thread.sleep(JAEGER_FLUSH_INTERVAL);   // TODO is this adequate?
+        } catch (InterruptedException e) {
+        }
+    }
+
+
+    /**
+     * Get all traces from the server
+     */
+    public List<JsonNode> getTraces() throws Exception {
+        return getTraces("");
+    }
+
+
+    /**
+     * Return all of the traces created since the start time given.  NOTE: The Jaeger Rest API
+     * requires a time in microseconds.  For convenience this method accepts milliseconds and converts.
+     *
+     * @param testStartTime in milliseconds
+     * @return
+     * @throws Exception
+     */
+    public List<JsonNode> getTracesSinceTestStart(long testStartTime) throws Exception {
+        List<JsonNode> traces = getTraces("start=" + (testStartTime * 1000));
+        return traces;
+    }
+
+
+    /**
+     * TODO: Figure out the Jaeger REST Api.  Key code can be found
+     *
+     *  https://github.com/uber/jaeger/blob/master/cmd/query/app/handler.go#L120-L130 with parameter info
+     *  https://github.com/uber/jaeger/blob/master/cmd/query/app/query_parser.go#L68-L81
+     *
+     * GET all traces for a service: http://localhost:3001/api/traces?service=something
+     * GET a Trace by id: http://localhost:3001/api/traces/23652df68bd54e15
+     * GET services http://localhost:3001/api/services
+     *
+     * GET after a specific time: http://localhost:3001/api/traces?service=something&start=1492098196598
+     * NOTE: time is in MICROseconds.
+     *
+     *
+     *
+     * @throws Exception
+     */
+    public List<JsonNode> getTraces(String parameters) throws Exception {
```

Comment:
@kevinearls we have to create separate Java REST client for Jaeger. We do not mess REST code with automation code.